vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -85,6 +108,18 @@ func (u *updater) RunOnce(ctx context.Context) {
 	timer := metrics_updater.NewExecutionTimer()
 	defer timer.ObserveTotal()
 
+	if u.useAdmissionControllerStatus {
+		s, err := u.statusClient.GetStatus()
```

Comment:
This flow actually has me thinking, shouldn't statusClient just expose IsStatusValid? Updater doesn't care if status is implemented as a Lease object and it shouldn't have to parse it.
If you hide this behind an interface with just one method, then in the tests you can use a mock object that just returns a boolean.

Reply:
I moved it to separated PR to not increase size of current PR: https://github.com/kubernetes/autoscaler/pull/2788